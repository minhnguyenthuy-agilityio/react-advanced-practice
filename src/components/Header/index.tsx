import { Avatar, Box, Flex, Icon, Text } from "@chakra-ui/react";

import { SearchBox } from "@/components";

import { USER } from "@/mocks";

import { ArrowDownIcon } from "@/assets/icons";

export const Header = () => {
  return (
    <Flex
      as="header"
      borderBottom="1px solid"
      borderColor="primary.400"
      px="55px"
      py="25px"
      justifyContent="space-between"
      alignItems="center"
    >
      <SearchBox />
      <Flex alignItems="center">
        <Box mr="20px" textAlign="right">
          <Text color="primary.700">{USER.name}</Text>
          <Text size="sm">{USER.country}</Text>
        </Box>
        <Avatar mr="10px" w="38px" h="38px" src={USER.avatar} />
        <Icon as={ArrowDownIcon} />
      </Flex>
    </Flex>
  );
};
