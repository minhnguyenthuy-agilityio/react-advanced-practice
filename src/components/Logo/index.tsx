import { Link as ChakraLink, Heading, Icon, Text } from "@chakra-ui/react";

import { Link as ReactRouterLink } from "react-router-dom";

import { LogoIcon } from "@/assets/icons";

import { ROUTES } from "@/constants";

export const Logo = () => (
  <Heading
    as="h1"
    py="35px"
    pl="20px"
    alignItems="center"
    borderBottom="solid 1px "
    borderColor="primary.400"
  >
    <ChakraLink
      display="flex"
      as={ReactRouterLink}
      to={ROUTES.TASKS_PROJECT_DEFAULT}
      _hover={{
        textDecoration: "none",
      }}
    >
      <Icon mr="8px" w="24px" h="24px" as={LogoIcon} />
      <Text color="primary.700" size="xm" fontWeight="semibold">
        Project M.
      </Text>
    </ChakraLink>
  </Heading>
);
