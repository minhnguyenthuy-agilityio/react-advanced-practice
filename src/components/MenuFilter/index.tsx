import { FC, memo } from "react";

import {
  Button,
  Icon,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
} from "@chakra-ui/react";

import { ArrowDownFilterIcon } from "@/assets/icons";

interface MenuFilterProps {
  buttonName: string;
  options: string[];
  icon?: FC<React.SVGProps<SVGSVGElement>>;
}

export const MenuFilter = memo(
  ({ buttonName, options, icon }: MenuFilterProps) => {
    return (
      <Menu>
        <MenuButton
          as={Button}
          variant="outline"
          leftIcon={icon && <Icon data-testid="icon-button-menu" as={icon} />}
          rightIcon={<Icon as={ArrowDownFilterIcon} />}
          w="122px"
          maxH="40px"
          fontWeight="medium"
        >
          {buttonName}
        </MenuButton>
        <MenuList>
          {options?.map((option, index) => (
            <MenuItem
              data-testid="menu-item"
              key={`${option}-${index}`}
              _hover={{
                bgColor: "secondary.200",
              }}
            >
              {option}
            </MenuItem>
          ))}
        </MenuList>
      </Menu>
    );
  },
);
