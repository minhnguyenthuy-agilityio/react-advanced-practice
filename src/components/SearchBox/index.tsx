import { ChangeEvent, KeyboardEvent, useState } from "react";

import { Icon, Input, InputGroup, InputLeftElement } from "@chakra-ui/react";

import { SearchIcon } from "@/assets/icons";

import { useTaskListStore } from "@/hooks";

export const SearchBox = () => {
  const [keyValue, setKeyValue] = useState("");
  const { setKeyword } = useTaskListStore();

  const handleOnChange = (
    event: ChangeEvent<
      HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
    >,
  ) => {
    setKeyValue(event.target.value);
  };

  const handleKeyDown = async (event: KeyboardEvent<HTMLInputElement>) => {
    if (event.key === "Enter") {
      setKeyword(keyValue.trim());
    }
  };

  return (
    <InputGroup bg="primary.200" borderRadius="6px" maxW="417px">
      <InputLeftElement left="12px" top="10%" fontSize="sm">
        <Icon as={SearchIcon} />
      </InputLeftElement>
      <Input
        pl="55px"
        h="full"
        placeholder="Search for anything..."
        py="13px"
        onChange={handleOnChange}
        onKeyDown={handleKeyDown}
      />
    </InputGroup>
  );
};
