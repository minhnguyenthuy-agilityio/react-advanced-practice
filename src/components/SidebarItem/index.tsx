import { Link as ChakraLink, Icon, Text } from "@chakra-ui/react";

import { DotIcon } from "@/assets/icons";

import { ROUTES } from "@/constants";

import { Link as ReactRouterLink } from "react-router-dom";

interface SidebarItemPropType {
  iconColor: string;
  title: string;
  projectId: string;
  isActive?: boolean;
}

export const SidebarItem = ({
  iconColor,
  title,
  projectId,
  isActive,
}: SidebarItemPropType) => {
  return (
    <ChakraLink
      as={ReactRouterLink}
      to={`${ROUTES.TASKS_PROJECT_DEFAULT}${projectId}`}
      display="flex"
      w="full"
      pl="14px"
      py="12px"
      borderRadius="6px"
      alignItems="center"
      background={isActive ? "secondary.200" : "transparent"}
      _hover={{
        textDecoration: "none",
        bgColor: "secondary.200",
      }}
    >
      <Icon mr="16px" as={DotIcon} color={iconColor} />
      <Text
        textOverflow="ellipsis"
        whiteSpace="nowrap"
        color={isActive ? "primary.700" : "primary.500"}
        fontWeight={isActive ? "semibold" : "medium"}
        overflowX="hidden"
      >
        {title}
      </Text>
    </ChakraLink>
  );
};
