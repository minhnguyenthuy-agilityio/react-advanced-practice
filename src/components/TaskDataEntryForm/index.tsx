import { useParams } from "react-router-dom";

import { useForm } from "react-hook-form";

import {
  Button,
  FormControl,
  FormErrorMessage,
  FormLabel,
  Input,
  Modal,
  ModalCloseButton,
  ModalContent,
  ModalHeader,
  ModalOverlay,
  Select,
  Stack,
  useToast,
} from "@chakra-ui/react";

import { TaskParam } from "@/interfaces";

import { PRIORITY_MAPPING, MESSAGES, Priority, Status } from "@/constants";

import { isValidImageUrl } from "@/helpers";

import { useMutationAddTask } from "@/hooks";

interface TaskDataEntryFormProps {
  isOpenForm: boolean;
  handleClose: () => void;
  defaultValues?: TaskParam;
}

const initialValues: TaskParam = {
  name: "",
  createdAt: new Date().toISOString(),
  priority: Priority.Low,
  projectId: "",
  status: Status.Todo,
  description: "",
  image: "",
};

export const TaskDataEntryForm = ({
  isOpenForm,
  handleClose,
  defaultValues,
}: TaskDataEntryFormProps) => {
  const {
    register,
    handleSubmit,
    reset,
    formState: { isSubmitting, errors },
  } = useForm<TaskParam>({
    defaultValues: defaultValues || initialValues,
  });
  const { mutate: mutateAdd, isSuccess, isError } = useMutationAddTask();
  const { projectId } = useParams();
  const toast = useToast();

  const titleForm = defaultValues ? "Edit Task" : "Create Task";

  const onCloseForm = () => {
    reset();
  };

  const onSubmit = (data: TaskParam) => {
    const taskNew = {
      ...data,
      projectId: projectId ? projectId : "pro00",
    };

    mutateAdd(taskNew);

    isError && toast({ title: MESSAGES.ADD_FAIL, status: "error" });

    if (isSuccess) {
      handleClose();
      reset();
      toast({
        title: MESSAGES.ADD_TASK_SUCCESS,
        status: "success",
      });
    }
  };

  return (
    <Modal size="xl" isOpen={isOpenForm} onClose={handleClose}>
      <ModalOverlay />
      <ModalContent textAlign="center" borderRadius="10px" as="div" p="30px">
        <ModalHeader p={10}>{titleForm}</ModalHeader>

        <ModalCloseButton onClick={onCloseForm} />

        <form onSubmit={handleSubmit(onSubmit)}>
          <Stack spacing={6}>
            <FormControl isInvalid={!!errors.name}>
              <FormLabel>Task Name</FormLabel>
              <Input
                fontSize="base"
                placeholder="Name"
                {...register("name", {
                  required: MESSAGES.FIELD_NAME_EMPTY,
                  minLength: {
                    value: 4,
                    message: MESSAGES.FIELD_NAME_LENGTH,
                  },
                })}
              />
              <FormErrorMessage>
                {errors.name && errors.name.message}
              </FormErrorMessage>
            </FormControl>

            <FormControl>
              <FormLabel>Description</FormLabel>
              <Input
                fontSize="base"
                placeholder="Here is description"
                {...register("description")}
              />
            </FormControl>

            <FormControl isInvalid={!!errors.priority}>
              <FormLabel>Priority</FormLabel>
              <Select
                color="primary.500"
                focusBorderColor="primary.500"
                fontSize="base"
                placeholder="Select priority"
                {...register("priority", {
                  required: MESSAGES.FIELD_PRIORITY_EMPTY,
                })}
              >
                {Object.entries(PRIORITY_MAPPING).map(
                  ([priorityKey, priorityValue]) => (
                    <option key={priorityKey} value={priorityKey}>
                      {priorityValue.label}
                    </option>
                  ),
                )}
              </Select>
              <FormErrorMessage>
                {errors.priority && errors.priority.message}
              </FormErrorMessage>
            </FormControl>

            <FormControl isInvalid={!!errors.image}>
              <FormLabel>Image</FormLabel>
              <Input
                fontSize="base"
                placeholder="http://"
                {...register("image", {
                  validate: (value) => {
                    if (!isValidImageUrl(value)) {
                      return MESSAGES.FIELD_IMAGE_INVALID;
                    }
                  },
                })}
              />
              <FormErrorMessage>
                {errors.image && errors.image.message}
              </FormErrorMessage>
            </FormControl>

            <Button
              mt={4}
              bgColor="primary"
              isLoading={isSubmitting}
              type="submit"
            >
              Submit
            </Button>
          </Stack>
        </form>
      </ModalContent>
    </Modal>
  );
};
