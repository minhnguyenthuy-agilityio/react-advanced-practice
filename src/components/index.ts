export * from "./Header";
export * from "./Sidebar";
export * from "./SearchBox";
export * from "./Logo";
export * from "./SidebarItem";
export * from "./TaskCard";
export * from "./TaskDataEntryForm";
export * from "./MenuFilter";
export * from "./TaskDetail";
export * from "./TaskList";
