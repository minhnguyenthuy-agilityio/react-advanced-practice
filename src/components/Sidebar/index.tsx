import { Box, Flex, Icon, Text, VStack } from "@chakra-ui/react";

import { useParams } from "react-router-dom";

import { Logo, SidebarItem } from "@/components";

import { PROJECTS } from "@/mocks";

import { TasksIcon } from "@/assets/icons";

export const SideBar = () => {
  const { projectId } = useParams();
  const project = projectId ? projectId : "";

  return (
    <Box as="aside" h="full" borderRight="1px solid" borderColor="primary.400">
      <Logo />
      <Box pr="10px" pl="2px">
        <Flex
          alignItems="center"
          pl="16px"
          py="14px"
          borderBottom="solid 1px"
          borderColor="primary.400"
        >
          <Icon mr="14px" as={TasksIcon} />
          <Text>Tasks</Text>
        </Flex>

        <VStack as="nav" spacing="2.5" alignItems="start">
          <Text
            pl="3"
            pt="7"
            pb="2"
            textTransform="uppercase"
            size="xs"
            fontWeight="700"
          >
            my projects
          </Text>
          {PROJECTS?.map(({ name, iconColor, projectId }, index) => (
            <SidebarItem
              key={`${name}-${index}`}
              iconColor={iconColor}
              title={name}
              projectId={projectId}
              isActive={project === projectId}
            />
          ))}
        </VStack>
      </Box>
    </Box>
  );
};
