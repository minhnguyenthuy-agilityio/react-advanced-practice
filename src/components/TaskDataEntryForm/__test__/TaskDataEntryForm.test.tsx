import { act, fireEvent, render, waitFor } from "@/helpers";

import { TaskDataEntryForm } from "@/components";

describe("TaskDataEntryForm component", () => {
  it("matchs snapshoot", () => {
    const element = render(
      <TaskDataEntryForm isOpenForm={true} handleClose={() => {}} />,
    );
    expect(element).toMatchSnapshot();
  });

  it("Validation for image URL", async () => {
    const { getByPlaceholderText, getByText } = render(
      <TaskDataEntryForm isOpenForm={true} handleClose={() => {}} />,
    );

    const imageInput = getByPlaceholderText("http://");
    const buttonSubmit = getByText("Submit");

    fireEvent.change(imageInput, {
      target: { value: "invalid" },
    });
    expect(imageInput).toHaveValue("invalid");

    act(() => {
      fireEvent.submit(buttonSubmit);
    });
    waitFor(() => {
      expect(getByText("Please enter a valid URL")).toBeInTheDocument();
    });
  });

  it("Validate required fields", async () => {
    const { queryByText, getByText } = render(
      <TaskDataEntryForm isOpenForm={true} handleClose={() => {}} />,
    );

    expect(queryByText("Name is required")).not.toBeInTheDocument();
    expect(queryByText("Please select priority")).not.toBeInTheDocument();

    act(() => {
      fireEvent.click(getByText("Submit"));
    });
    waitFor(() => {
      expect(getByText("Name is required")).toBeInTheDocument();
      expect(getByText("Please select priority")).toBeInTheDocument();
    });
  });

  it("renders form successfully", () => {
    const { getByText } = render(
      <TaskDataEntryForm isOpenForm={true} handleClose={() => {}} />,
    );

    expect(getByText("Task Name")).toBeInTheDocument();
    expect(getByText("Description")).toBeInTheDocument();
    expect(getByText("Priority")).toBeInTheDocument();
    expect(getByText("Image")).toBeInTheDocument();
  });

  it("should reset form when called", () => {
    const handleClose = jest.fn();
    const { getByLabelText } = render(
      <TaskDataEntryForm isOpenForm={true} handleClose={handleClose} />,
    );

    fireEvent.change(getByLabelText("Task Name"), {
      target: { value: "New Task" },
    });
    fireEvent.change(getByLabelText("Description"), {
      target: { value: "Task description" },
    });
    fireEvent.change(getByLabelText("Priority"), { target: { value: "High" } });
    fireEvent.change(getByLabelText("Image"), {
      target: { value: "https://example.com/image.jpg" },
    });

    fireEvent.click(getByLabelText("Close"));

    expect(getByLabelText("Task Name")).toHaveValue("");
    expect(getByLabelText("Description")).toHaveValue("");

    expect(handleClose).toHaveBeenCalled();
  });
});
