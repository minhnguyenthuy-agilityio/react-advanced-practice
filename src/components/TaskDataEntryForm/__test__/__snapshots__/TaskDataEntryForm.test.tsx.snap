// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDataEntryForm component matchs snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    >
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-hidden="true"
        class="chakra-modal__overlay css-hdd9l7"
        data-aria-hidden="true"
        style="opacity: 0;"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="chakra-modal__content-container css-1u2cvaz"
          tabindex="-1"
        >
          <div
            animate="enter"
            aria-labelledby="chakra-modal--header-:r0:"
            aria-modal="true"
            class="chakra-modal__content css-dpak00"
            custom="[object Object]"
            exit="exit"
            id="chakra-modal-:r0:"
            initial="exit"
            role="dialog"
            tabindex="-1"
            variants="[object Object]"
          >
            <header
              class="chakra-modal__header css-u0i2nk"
              id="chakra-modal--header-:r0:"
            >
              Create Task
            </header>
            <button
              aria-label="Close"
              class="chakra-modal__close-btn css-1ik4h6n"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <form>
              <div
                class="chakra-stack css-ahthbn"
              >
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-g6pte"
                    for="field-:r3:"
                    id="field-:r3:-label"
                  >
                    Task Name
                  </label>
                  <input
                    class="chakra-input css-1v0zw5w"
                    id="field-:r3:"
                    name="name"
                    placeholder="Name"
                  />
                </div>
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-g6pte"
                    for="field-:r4:"
                    id="field-:r4:-label"
                  >
                    Description
                  </label>
                  <input
                    class="chakra-input css-1v0zw5w"
                    id="field-:r4:"
                    name="description"
                    placeholder="Here is description"
                  />
                </div>
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-g6pte"
                    for="field-:r5:"
                    id="field-:r5:-label"
                  >
                    Priority
                  </label>
                  <div
                    class="chakra-select__wrapper css-12edts4"
                  >
                    <select
                      class="chakra-select css-n2lck0"
                      id="field-:r5:"
                      name="priority"
                    >
                      <option
                        value=""
                      >
                        Select priority
                      </option>
                      <option
                        value="0"
                      >
                        Low
                      </option>
                      <option
                        value="1"
                      >
                        High
                      </option>
                    </select>
                    <div
                      class="chakra-select__icon-wrapper css-k3889b"
                    >
                      <svg
                        aria-hidden="true"
                        class="chakra-select__icon"
                        focusable="false"
                        role="presentation"
                        style="width: 1em; height: 1em; color: currentColor;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="chakra-form-control css-1kxonj9"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-g6pte"
                    for="field-:r6:"
                    id="field-:r6:-label"
                  >
                    Image
                  </label>
                  <input
                    class="chakra-input css-1v0zw5w"
                    id="field-:r6:"
                    name="image"
                    placeholder="http://"
                  />
                </div>
                <button
                  class="chakra-button css-ilx49x"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
